app/state_machines/form_state_machine.rb

```diff
@@ -0,0 +1,60 @@
+module FormStateMachine
+  extend ActiveSupport::Concern
+
+  included do
+    include AASM
+
+    enum :state, {
+      draft: "draft",
+      live: "live",
+      draft_live: "draft_live",
+      archived: "archived",
+      draft_archived: "draft_archived",
+    }
```

Comment:
we can but we were trying to keep it as close to what users see when they view list of statuses

what do others thing about @DavidBiddle  suggestion? @alphagov/forms-devs

Reply:
I also had comments about the transition names, but I put them on the other PR 😅 

https://github.com/alphagov/forms-api/pull/422#discussion_r1466117774

Reply:
Not a developer, but this seems like a good idea. It reads more explicit to me. 
And in terms of what the user sees this is basically how they will see it. They don’t know that it’s a single state - they’ll see it as 2 separate things, a draft AND a live/archived form

Reply:
@thomasiles @jamie-o-wilkinson  and myself had a chat about this is and  were concerned about about how form-admin might map `live_with_draft` state to two different states for the users but it turns up this shouldn't be a problem because form-api forms model `has_draft_version` and `has_live_version` method which forms-admin uses to output the two different tags. we might want to change this in future but for the time being we happy to go with `live_with_draft` state. see implementation in https://github.com/alphagov/forms-api/pull/422

Reply:
```mermaid
stateDiagram-v2

    [*] --> Form

    state Form {
        [*] --> Draft
        Draft --> [*]
        --
        Archived --> Live
        Live --> Archived
    }

    Draft --> Live
```